### PR TITLE
Fix #244: Response codes for exceptions

### LINKED ISSUE #244: Graceful uploading error handling
For now application [doesn't](https://github.com/refile/refile/blob/master/lib/refile/app.rb#L64) handle backend fail cases and just responds with 500 error code, which is not convenient. So maybe it would be nice to handle those exceptions in app and return 413 in case of size validation, or 400 when ArgumentError is thrown. It'd improve semantics and probably help handling errors on frontend.

### PR DESCRIPTION
**Pull request by toothfairy**
Handles uploading exceptions and responds with appropriate status

### CODE REVIEW

File: lib/refile/app.rb [hunk 1]
```diff
@@ -85,6 +85,16 @@ class App < Sinatra::Base
       "forbidden"
     end
 
+    error ArgumentError do
```
Comment (jnicklas) [inline lib/refile/app.rb]: ArgumentError is a very general error to rescue. I think it's not ideal to assume that any particular thing happened given an ArgumentError.

Comment (toothfairy) [inline lib/refile/app.rb]: @jnicklas should it raise different kind of exception (ie Refile::InvalidFile or UnreadableFile) in backend macros?

Comment (jnicklas) [inline lib/refile/app.rb]: That's a good idea probably. Yes.

**Comment by jnicklas** [comment]
Nice :) Thanks!

**Comment by toothfairy** [comment]
:dancer:

**Pull request merged**